### DISCUSSION TITLE
How to define the boundaries of 2D slice cut from 3D model

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#25981] [Q&A General]
Dear MOOSEers,

I run a 3D simulation first, and cut a cross-section using '**slice**' function in paraview to get a 2D model. Then I want to run simulation on the 2D slice with mesh of the slice using **FileMesh**, but how to define the boundaries of it?  I tried to use boundaries such as "left right..." directly, but the error pop up as follow. Thanks.

> the following node sets (ids) do not exist on the mesh: left (1), right (2)

**Comment by richmondodufisan** [comment]
If you know how to write the geometry of the slice mathematically you can use ParsedGenerateSideset with FileMeshGenerator. You can extract whatever postprocessor value you need from there. You might also be able to run a simulation only on that slice by defining it as a block with SubdomainBoundingBoxGenerator (I haven't tried this, but give it a shot)

```
[Mesh]
  [name_1]
    type = FileMeshGenerator
    file = <meshfile>
  []
  [name_2]
    type = ParsedGenerateSideset
    input = name_1
    combinatorial_geometry =<parsed expression for slice e.g '((z > 0-1e-8) & (z < 0+1e-8))' for a slice at z = 0>
    new_sideset_name =<what_you_want_to_call it>
  []
[]
```

**Reply by rh201** [reply]
thanks. @richmondodufisan

**Reply by GiudGiud** [reply]
Hello

Once you have extracted a slice using this technique, you can turn it into blcoks with this:
https://mooseframework.inl.gov/source/meshgenerators/LowerDBlockFromSidesetGenerator.html

then if you want to delete the rest use this first to create an independent mesh:
https://mooseframework.inl.gov/source/meshgenerators/BlockToMeshConverterGenerator.html

Guillaume
